src/test/mir-opt/issue-91633.rs

```diff
@@ -0,0 +1,16 @@
+// EMIT_MIR issue_91633.hey.mir_map.0.mir
+fn hey<T> (it: &[T])
+ where
+     [T] : std::ops::Index<usize>,
+ {
+     let _ = &it[0];
+ }
+
+// EMIT_MIR issue_91633.bar.mir_map.0.mir
+fn bar<T> (it: Box<[T]>)
```

Comment:
worth adding a tests where we do expect an index operator to be generated (e.g. the "hey" and "bar" functions without a where clause)